### DISCUSSION TITLE
Replace a texture by a new one

### DISCUSSION THREAD

**Discussion by Makio64** [donmccurdy/glTF-Transform#1229] [General]
How to replace a texture by another one, for example replace an image texture by an uastc texture in all the model?
Should we use setImage / setExtension / setMimeType on the current one ?

**Comment by donmccurdy** [comment]
Yes, you can use any of the methods found in the [Texture class](https://gltf-transform.dev/modules/core/classes/Texture). If you're not changing the format of the image, then `.setImage(...)` with the new image (Buffer or Uint8Array) is enough. If the format changes then updating the URI and MIME type would also be required. When using texture formats that require a glTF extension, like KTX2, WebP, or AVIF, you'll need to register and create the relevant extension as shown in:

https://gltf-transform.dev/modules/extensions/classes/KHRTextureBasisu

For example:

```js
import { KHRTextureBasisu } from '@gltf-transform/extensions';

io.registerExtensions([KHRTextureBasisu]);

document.createExtension(KHRTextureBasisu).setRequired(true);
```

**Reply by Makio64** [reply]
Thanks it works perfectly with setImage and setMimeType + setRequired on the extension :)
